### ISSUE TITLE
Bug: RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by HarelM** [eslint/eslint#20099]
### Environment

Node version: 22.13
npm version: 10.9.2
Local ESLint version:
Global ESLint version:
Operating System: Linux/Mac

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';
import reactPlugin from 'eslint-plugin-react';
import reactHooksPlugin from 'eslint-plugin-react-hooks';
import reactRefreshPlugin from 'eslint-plugin-react-refresh';

export default tseslint.config({
  extends: [
    eslint.configs.recommended,
    tseslint.configs.recommended,
  ],
  files: ['**/*.{js,jsx,ts,tsx}'],
  ignores: [
    "dist/**/*",
  ],
  languageOptions: {
    ecmaVersion: 2024,
    sourceType: 'module',
    globals: {
      global: 'readonly'
    }
  },
  settings: {
    react: { version: '18.2' }
  },
  plugins: {
    'react': reactPlugin,
    'react-hooks': reactHooksPlugin,
    'react-refresh': reactRefreshPlugin
  },
  rules: {
    'react-refresh/only-export-components': [
      'warn',
      { allowConstantExport: true }
    ],
    "@typescript-eslint/no-explicit-any": "off",
    '@typescript-eslint/no-unused-vars': [
      'warn',
      {
        varsIgnorePattern: '^_',
        caughtErrors: 'all',
        caughtErrorsIgnorePattern: '^_',
        argsIgnorePattern: '^_'
      }
    ],
    'no-unused-vars': 'off',
    'react/prop-types': 'off',
    'no-undef': 'off',
    'indent': ['error', 2],
    'no-var': 'error',
    '@typescript-eslint/no-non-null-asserted-optional-chain': 'off',
    '@typescript-eslint/ban-ts-comment': 'off',
    '@typescript-eslint/no-empty-object-type': 'off',

  },
  linterOptions: {
    reportUnusedDisableDirectives: true,
    noInlineConfig: false
  }
}
)

```
</details>

```js
eslint
```

### What did you expect to happen?

lint should pass.

### What actually happened?

```
ESLint: 9.35.0

RangeError: Maximum call stack size exceeded
Occurred while linting /Users/runner/work/maputnik/maputnik/src/components/AppMessagePanel.tsx:1
Rule: "indent"
    at WeakMap.has (<anonymous>)
    at OffsetStorage.getDesiredIndent (/Users/runner/work/maputnik/maputnik/node_modules/eslint/lib/rules/indent.js:427:33)
    at OffsetStorage.getDesiredIndent (/Users/runner/work/maputnik/maputnik/node_modules/eslint/lib/rules/indent.js:444:11)
    at OffsetStorage.getDesiredIndent (/Users/runner/work/maputnik/maputnik/node_modules/eslint/lib/rules/indent.js:444:11)
    at OffsetStorage.getDesiredIndent (/Users/runner/work/maputnik/maputnik/node_modules/eslint/lib/rules/indent.js:444:11)
    at OffsetStorage.getDesiredIndent (/Users/runner/work/maputnik/maputnik/node_modules/eslint/lib/rules/indent.js:444:11)
    at OffsetStorage.getDesiredIndent (/Users/runner/work/maputnik/maputnik/node_modules/eslint/lib/rules/indent.js:444:11)
    at OffsetStorage.getDesiredIndent (/Users/runner/work/maputnik/maputnik/node_modules/eslint/lib/rules/indent.js:444:11)
    at OffsetStorage.getDesiredIndent (/Users/runner/work/maputnik/maputnik/node_modules/eslint/lib/rules/indent.js:444:11)
    at OffsetStorage.getDesiredIndent (/Users/runner/work/maputnik/maputnik/node_modules/eslint/lib/rules/indent.js:444:11)
```

### Link to Minimal Reproducible Example

https://github.com/maplibre/maputnik/actions/runs/17604824640/job/50013432290?pr=1301

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I was trying to update typescript to latest version (5.9.3), so it might not be a problem with this repo but rather one of the plugins, but I don't know how to verify that.

The following PR is where I see this issue:
https://github.com/maplibre/maputnik/pull/1301

I'm not sure I know how to create a minimal reproduction, but it's easy to clone the repo and see the problem when upgrading typescript to version 5.9.3.

Let me know if I can help in any way.

**Comment by vilsongabriel** [comment]
I’m also experiencing this issue with TypeScript 5.9.2. After downgrading to 5.8.3, the error does not occur.

**Comment by kirkwaiblinger** [comment]
FYI this affected `@stylistic/indent` in https://github.com/eslint-stylistic/eslint-stylistic/issues/915 , see also https://github.com/typescript-eslint/typescript-eslint/issues/11455

**Comment by kirkwaiblinger** [comment]
TL;DR TS 5.9 changed the underlying tokens for jsx fragments and `indent` struggles with the new version. typescript-eslint might still make a change to override TS's token output for downstream consumers. But note that `@stylistic/indent` is the actively maintained version of `indent`, not ESLint core, and I think a fix has already been released so that they handle the new token structure.

**Comment by kirkwaiblinger** [comment]
Minimal repro: https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=MYewdgzgLgBAZgJwIYHMC2BTMsC8MA8AfAFD4D0hQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQJYDsATRfAF3SgHcBDafScMAXxCaA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

**Comment by mdjermanovic** [comment]
As noted, the core [`indent`](https://eslint.org/docs/latest/rules/indent) rule is deprecated, thus no longer maintained, and you should switch to [`@stylistic/indent`](https://eslint.style/rules/indent) or a code formatter like Prettier.

Still, I'll leave this issue open for visibility until the change in tokens that caused this is reverted (https://github.com/microsoft/TypeScript/pull/62425), as it may be affecting other rules.

(@kirkwaiblinger thanks for the context!)

**Comment by Revanwjy-alt** [comment]
main disini cuan

jo777.help

**Comment by Revanwjy-alt** [comment]
main disni berhadiah menarik

jo777.help

**Comment by aladdin-add** [comment]
closing as https://github.com/microsoft/TypeScript/pull/62425 has been merged. please file a new issue if you are still encountering the issue.
